packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -104,9 +105,43 @@ contract L2Genesis is Deployer {
     /// @notice The address of the deployer account.
     address internal deployer;
 
+    // celo - create and write predeploy map
+    mapping(string => address) public deployedContractNamesToAddresses;
+    string internal _celoL2Outfile;
+
+    function celoL2Outfile() internal view returns (string memory _env) {
+        _env = vm.envOr(
+            "L2_OUTFILE",
+            string.concat(vm.projectRoot(), "/deployments/", vm.toString(block.chainid), "-l2-deploy.json")
+        );
+    }
+
+    function celoSave(string memory _name, address _impl, address _proxy) public {
```

Comment:
Should the name include "celo"? As far as I can tell it only writes the addresses of OP L2 contracts, not the celo specific ones. SO maybe we just use generic l2 names?

And is there a reason we don't write the celo specific contract information as well?

Reply:
I added this prefix to function names so that it was clear where the changes came from when dealing with future upstream merges / merge conflicts. Additionally the `L2Genesis` contract inherits the `Artifacts` abstract contract which already defines functions like `save` with incompatible hard coded values. 

Celo specific contracts weren't included due to the [l2 testnet config not deploying celo contracts by default](https://github.com/celo-org/optimism/blob/celo7/.envrc.example#L75) when I made this.

I'm open to any suggestions or alternatives :)

Reply:
Makes sense. I have no strong opinions either. We can always improve it later.